### PR TITLE
Summary WIP

### PR DESCRIPTION
**Pull request by joanofdart**
WIP

### CODE REVIEW

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 1]
```diff
@@ -1,13 +1,185 @@
...
+    <p description class="mat-body-1">
+      Please verify that all the information you have entered is correct before submitting to Archipel Content for validation.
+    </p>
+
+    <section id="main-information">
+      <h3>Main Information</h3>
+      <article>
+        <h5>
+          Reference
+          <a mat-icon-button [routerLink]="getPath('main')"><mat-icon svgIcon="pencil" color="primary"></mat-icon></a>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:14]: Would be nice to go to the right fragment (#id) when you click on the link.

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 2]
```diff
@@ -1,13 +1,185 @@
...
+        </h5>
+        <ul>
+          <li>
+            <strong>International Title </strong>
+            <missing-control [control]="title.get('international')" [link]="getPath('main')"></missing-control>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:19]: Same for the link.

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 3]
```diff
@@ -1,13 +1,454 @@
...
+
+    <section id="production-information">
+      <h3>Production Information</h3>
+      <article>
+        <h5>
+          Production Companies
+          <a mat-icon-button [routerLink]="getPath('production')"><mat-icon svgIcon="pencil" color="primary"></mat-icon></a>
+        </h5>
+        <ul>
+          <ng-container *ngFor="let producer of form.stakeholders.get('productionCompany').controls; let isLast = last">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:214]: Why not using "li" ?

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 4]
```diff
@@ -1,13 +1,454 @@
...
+            <missing-control [control]="producer.get('displayName')" [link]="getPath('production')"></missing-control>
+            <ng-container *ngIf="producer.get('countries').value.length">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:216]: Change to : 
```html
<span *ngIf="producer.get('countries').value.length">
  ({{ producer.get('countries').value | translateSlug: 'TERRITORIES' }})
</span >
```

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 5]
```diff
@@ -1,13 +1,454 @@
...
+          </ng-container>
+        </ul>
+      </article>
+
+      <article>
+        <h5>Co-production Companies
+          <a mat-icon-button [routerLink]="getPath('production')"><mat-icon svgIcon="pencil" color="primary"></mat-icon></a>
+        </h5>
+        <ul>
+          <ng-container *ngFor="let coProducer of form.stakeholders.get('coProductionCompany').controls; let isLast = last">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:233]: Same as above : 
```html
<ul>
  <li *ngFor="let coProducer of form.stakeholders.get('coProductionCompany').controls">
    <span  *ngIf="coProducer.get('countries').value.length">
      ({{ coProducer.get('countries').value | translateSlug: 'TERRITORIES' }})
    </span>
  </li>
</ul>
```

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 6]
```diff
@@ -1,13 +1,454 @@
...
+            <ng-container *ngIf="producerHasNoValue(producer) else displayProducer">
+              <a [routerLink]="getPath('production')" class="missing-control">
+                Missing
+              </a>
+            </ng-container>
+            <ng-template #displayProducer>
+              <missing-control [control]="producer.get('firstName')" [link]="getPath('production')"></missing-control>
+              <span>&nbsp;</span>
+              <missing-control [control]="producer.get('lastName')" [link]="getPath('production')"></missing-control>
+              <span *ngIf="producer.get('role').value"> (</span>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:263]: Maybe you could update the missing-control to put directly the value into parenthesises is this happens all the time.

File: libs/movie/src/lib/movie/pages/summary/summary.component.html (hunk 5 above)
Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/pages/summary/summary.component.html:233]: Same for below.

File: libs/movie/src/lib/movie/pages/summary/summary.component.ts [hunk 7]
```diff
@@ -31,6 +32,30 @@ export class MovieFormSummaryComponent {
...
+
+  public producerHasNoValue(producer: CreditForm) {
+    return !producer.get('firstName').value || !producer.get('lastName').value || !producer.get('role').value;
+  }
+
+  public festivalPrizeHasNoValue(festivalPrize: MoviePrizeForm) {
+    return !festivalPrize.get('name').value && !festivalPrize.get('prize').value && !festivalPrize.get('year').value && !festivalPrize.get('premiere').value;
+  }
+
+  public reviewHasNoValue(review: MovieReviewForm) {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/summary/summary.component.ts:47]: Maybe using `review.valid` would be enough ?

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 8]
```diff
@@ -1,13 +1,454 @@
...
+          <ng-container *ngFor="let review of form.review.controls">
+            <li>
+              <strong>Festivals </strong>
+              <ng-container *ngIf="reviewHasNoValue(review) else display">
+                <a [routerLink]="getPath('reviews')" class="missing-control">
+                  Missing
+                </a>
+              </ng-container>
+              <ng-template #display>
+                <missing-control [control]="review.get('criticName')" [link]="getPath('reviews')">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:420]: You're already checking if values exist above. You can do : 
```html
<ng-template #display *ngIf="reviewControl.value as review">
  <span>{{ review.criticName }}, {{ review.journalName }}, {{ review.criticQuote }}, {{ review.revueLink }}</span>
</ng-template>
```
In this case I renamed "review" to "reviewControl".

File: libs/movie/src/lib/movie/pages/summary/summary.component.ts (hunk 7 above)
Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/pages/summary/summary.component.ts:47]: Same for the other one.

File: libs/movie/src/lib/movie/pages/summary/summary.component.html (hunk 6 above)
Comment (joanofdart) [reply libs/movie/src/lib/movie/pages/summary/summary.component.html:263]: Its not used that much imo.

File: libs/movie/src/lib/movie/pages/summary/summary.component.ts (hunk 7 above)
Comment (joanofdart) [reply libs/movie/src/lib/movie/pages/summary/summary.component.ts:47]: removed all these funcs and just did form.valid (form = each form)

**Pull request closed without merging**